src/workspaces/features/SidebarMenu/SidebarMenu.tsx

```diff
@@ -44,29 +54,58 @@ const SidebarMenu = (props: SidebarMenuProps) => {
     modifiers: POPPER_MODIFIERS,
   });
   const innerMenuRef = useRef<HTMLDivElement>(null);
+
   useOnClickOutside(innerMenuRef, () => {
-    setFalse();
+    if (!isDialogOpen) {
+      // Do not close the menu if the user click in the dialog
+      setFalse();
+    }
   });
+
   useEffect(() => {
     if (isOpen) {
       setFalse();
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [router.asPath]);
 
+  const { data, loading, refetch } = useQuery<
+    WorkspacesPageQuery,
+    WorkspacesPageQueryVariables
+  >(WorkspacesPageDocument, {
+    variables: { page: 1, perPage: 5 },
+    fetchPolicy: "no-cache",
+  });
+
+  useCacheKey("workspaces", () => {
+    refetch();
+  });
+
+  const showMore = (perPage: number) => {
+    refetch({
+      page: 1,
+      perPage,
+    });
+  };
+
+  if (!workspace) {
+    return null;
+  }
+
   return (
     <div className="w-full" ref={innerMenuRef}>
       <button
         className="group flex h-16 w-full items-center bg-gray-800 px-2 text-left hover:bg-gray-600"
         ref={setReferenceElement}
         onClick={toggle}
       >
-        {workspace.country && (
+        {workspace.countries && workspace.countries.length === 1 && (
```

Comment:
Why `countries` would not be provided ?

Reply:
It's an unnecessary additional check. Will remove it